### PR TITLE
refactor: Refactored controllers to use DI library

### PR DESCRIPTION
**Pull request by franciscog-kainos**
This PR just adds DI to the project.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,8 +41,8 @@
     "gulp-watch-sass": "^1.4.1",
     "node-sass": "^4.13.1",
     "sass-lint": "^1.13.1",
-    "tslint": "^6.0.0",
-    "typescript": "^3.7.5",
+    "tslint": "^5.7.0",
```
Comment (geek-soft) [inline package.json]: Please revert that change please. I suggest we use latest versions.

File: package.json [hunk 2]
```diff
@@ -34,8 +41,8 @@
...
+    "typescript": "^3.5.3",
```
Comment (geek-soft) [inline package.json]: Same here

File: src/controllers/LandingController.ts [hunk 3]
```diff
@@ -1,6 +1,11 @@
 import { Request, Response, NextFunction } from 'express';
+import { controller, httpGet } from 'inversify-express-utils';
 
+@controller('/')
 export class LandingController {
 
-    public renderView = (req: Request, res: Response, next: NextFunction) => res.render('landing');
+    @httpGet('')
```
Comment (geek-soft) [inline src/controllers/LandingController.ts]: Out of curiosity is that empty string required or can that arg be omitted?

Comment (franciscog-kainos) [reply src/controllers/LandingController.ts]: It's required :(

File: tsconfig.json [hunk 4]
```diff
@@ -10,6 +10,8 @@
       "emitDecoratorMetadata": true,
       "moduleResolution": "node",
       "importHelpers": true,
+      "lib": ["es6"],
+      "types": ["reflect-metadata"],
```
Comment (geek-soft) [inline tsconfig.json]: After reading https://www.typescriptlang.org/v2/en/tsconfig#types I suggest we remove it. It narrows down what types are loaded if I understand correctly.

File: src/views/layout.njk [hunk 5]
```diff
@@ -82,6 +82,6 @@
 
 {% block bodyEnd %}
   {# Run JavaScript at end of the <body>, to avoid blocking the initial render. #}
-  <script src="govuk/all.js"></script>
+  <script src="/govuk/all.js"></script>
```
Comment (geek-soft) [inline src/views/layout.njk]: Revert to keep it focused on DI. Can be done in PR for user story.

File: src/Server.ts [hunk 6]
```diff
@@ -1,46 +1,58 @@
...
-    this.app.listen(this.app.get('port'), () => {
-      console.log(('  App is running at http://localhost:%d in %s mode'), this.app.get('port'), this.app.get('env'));
+    this.server.build().listen(this.port, () => {
+      console.log(('  App is running at http://localhost:%d in %s mode'), this.port, process.env.NODE_ENV);
       console.log('  Press CTRL-C to stop\n');
     });
   }
 
-  private setupStaticFolders(app: express.Express): void {
+  private createContainerWithBindings(): Container {
```
Comment (geek-soft) [inline src/Server.ts]: Good for now but in the future it should be moved away from server.ts to reduce reasons for change.

File: src/Types.ts [hunk 7]
```diff
@@ -0,0 +1,2 @@
+export const TYPES = {
```
Comment (n-r-b) [inline src/Types.ts]: do we need to add this right now?

**Pull request merged**